### PR TITLE
fix(raiden): cleanup reconnectTimer on shutdown

### PR DESCRIPTION
**Pull request by ghost**
This commit fixes a bug where the Raiden reconnect timer would not cleanup when triggering a graceful shutdown.

In addition, I have already fixed the issue in my other feature branch where I moved the cleanup logic into the BaseClient.

### CODE REVIEW

**Review by sangaman — APPROVED** [review]

**Comment by sangaman** [comment]
Looks good. I think it'd be nice (but not necessary) to have the reconnection timer cleanup logic in the base client since. Do you want to just merge this for now and then refactor it in your next branch? I figure you could handle that in this PR if it's not any trouble.

**Comment by erkarl** [comment]
Good idea. This will make the other PR smaller and easier to review. I've moved the reconnect cleanup logic into the abstract `BaseClient`.

File: lib/BaseClient.ts [hunk 1]
```diff
@@ -45,6 +45,13 @@ abstract class BaseClient extends EventEmitter {
   public isDisconnected(): boolean {
     return this.status === ClientStatus.Disconnected;
   }
+  public async close() {
+    if (this.reconnectionTimer) {
+      clearTimeout(this.reconnectionTimer);
+    }
+    await this.stop();
+  }
+  protected abstract stop(): Promise<void>;
```
Comment (sangaman) [inline lib/BaseClient.ts]: Why have this return a promise if we're not doing anything asynchronous with it?

File: lib/lndclient/LndClient.ts [hunk 2]
```diff
@@ -340,15 +340,12 @@ class LndClient extends BaseClient {
       });
   }
 
-  /** Ends all subscriptions and reconnection attempts. */
-  public close = () => {
+  /** Ends all subscriptions attempts. */
```
Comment (sangaman) [inline lib/lndclient/LndClient.ts]: Maybe have this comment say something along the lines of "Lnd specific cleanup" (like you've done for Raiden) and move this general description to the base `close` method. Also "subscriptions attempts" looks like a typo here, maybe best to say something like "Ends all connections, subscriptions, and timers for for this client."

File: lib/BaseClient.ts (hunk 1 above)
Comment (sangaman) [reply lib/BaseClient.ts]: Another thought I have here is that "close" and "stop" might get confused with each other, maybe it'd be more descriptive to call this method "closeSpecific" or something like that?

Comment (erkarl) [reply lib/BaseClient.ts]: > Why have this return a promise if we're not doing anything asynchronous with it?
I was assuming we're going to have async cleanup in the future, but you're right - it's not necessary right now. Removed it.

> maybe it'd be more descriptive to call this method "closeSpecific"
`closeSpecific` sounds great. Changed.

**Review by sangaman — APPROVED** [review]
Looks good, thanks! Just squash the commits then merge.

**Pull request merged**